modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -517,19 +521,37 @@ class ApiClient(object):
         else:
             return content_types[0]
 
-    def update_params_for_auth(self, headers, querys, auth_settings):
+    def update_params_for_auth(self, headers, querys, auth_settings,
+                               resource_path, method, body):
         """Updates header and query params based on authentication setting.
 
         :param headers: Header parameters dict to be updated.
         :param querys: Query parameters tuple list to be updated.
         :param auth_settings: Authentication setting identifiers list.
+        :resource_path: A string representation of the HTTP request resource path.
+        :method: A string representation of the HTTP request method.
+        :body: A string representation of the body of the HTTP request.
         """
         if not auth_settings:
             return
 
         for auth in auth_settings:
             auth_setting = self.configuration.auth_settings().get(auth)
             if auth_setting:
+{{#hasHttpSignatureMethods}}
+                if auth_setting['type'] == 'http-signature':
```

Comment:
Is this a specific if statement that can be included in the below if/elif case handling?
How about
- changing this if statement to `if auth_setting['in'] == 'header' and auth_setting['type'] == 'http-signature':`
- deleting the continue line from the bottom of this if block
- change `if auth_setting['in'] == 'cookie':` to `elif auth_setting['in'] == 'cookie':`

Reply:
what you are proposing would not work exactly as is, because the block is generated conditionally. But I get the idea.

Reply:
Hmm, it's a bit tricky to make it clean because of the indentation requirements. For example, we could have something like below but I feel it makes the logic harder to understand.

```
            if auth_setting:
                if auth_setting['in'] == 'cookie':
                    headers['Cookie'] = auth_setting['value']
                elif auth_setting['in'] == 'header':
{{#hasHttpSignatureMethods}}
                    if auth_setting['type'] == 'http-signature':
                        # The HTTP signature scheme requires multiple HTTP headers
                        # that are calculated dynamically.
                        signing_info = self.configuration.signing_info
                        if signing_info is None:
                            raise Exception("HTTP signature configuration is missing")
                        auth_headers = signing_info.get_http_signature_headers(
                                            resource_path, method, headers, body, querys)
                        for key, value in auth_headers.items():
                            headers[key] = value
                    else:
                        headers[auth_setting['key']] = auth_setting['value']
{{/hasHttpSignatureMethods}}
{{^hasHttpSignatureMethods}}
                    headers[auth_setting['key']] = auth_setting['value']
{{/hasHttpSignatureMethods}}
                elif auth_setting['in'] == 'query':
                    querys.append((auth_setting['key'], auth_setting['value']))
                else:
                    raise ApiValueError(
                        'Authentication token must be in `query` or `header`'
                    )

Reply:
That's true. You cold change this block to an elif then and move it to be conditionally included down below.

Reply:
You don't need all that code, how about:
```
            if auth_setting:
                if auth_setting['in'] == 'cookie':
                    headers['Cookie'] = auth_setting['value']
                elif auth_setting['in'] == 'header':
                    if auth_setting['type'] != 'http-signature':
                        headers[auth_setting['key']] = auth_setting['value']
{{#hasHttpSignatureMethods}}
                    else:
                        # The HTTP signature scheme requires multiple HTTP headers
                        # that are calculated dynamically.
                        signing_info = self.configuration.signing_info
                        # NOTE: no need to raise an exception because this entry only exists if signng_info is not None
                        auth_headers = signing_info.get_http_signature_headers(
                                            resource_path, method, headers, body, querys)
                        headers.update(auth_headers)
{{/hasHttpSignatureMethods}}
                elif auth_setting['in'] == 'query':
                    querys.append((auth_setting['key'], auth_setting['value']))
                else:
                    raise ApiValueError(
                        'Authentication token must be in `query` or `header`'
                    )
```

Reply:
done